jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -192,8 +189,20 @@ static String getPathInContext(Request request)
      */
     HttpFields getTrailers();
 
+    /**
+     * @return The timestamp that the request was received/created in milliseconds, relative to the "wall clock" given
+     * by {@link System#currentTimeMillis()}.
+     */
     long getTimeStamp();
 
+    /**
+     * @return The timestamp that the request was received/created in nanoseconds, relative to {@link System#nanoTime()}.
+     */
+    default long getNanoTimeStamp()
+    {
+        return System.nanoTime() - TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis() - getTimeStamp());
```

Comment:
It probably should be used by the stats handler, although that is probably getting it directly from the nano timestamp in the HttpStream.
When we really have a channel/stream, then the actual implementation is :
```java

            HttpStream stream = _httpChannel.getHttpStream();
            if (stream != null)
                return stream.getNanoTimeStamp();
```

This in turn calls `NanoTime.now()`. So this impl is only for test requests.

The method exists because we want to encourage use of nano time for latencies measured from the first time the request is received.   It is different to the `getTimeStamp` method which is returning the wall clock.

The stats handler should use this rather than create its own timestamp that is someway through the handler chain.

I'll improve javadoc.